Hide pretty-print button from mapped files

jasonLaster opened the pull request:
Fixes #420

jlongster commented on public/js/utils/source.js:
```diff
@@ -29,6 +29,11 @@ function isJavaScript(url, contentType = "") {
          contentType.includes("javascript");
 }
 
+function isGenerated(source) {
```
This check seems wrong. A "generated" source is considered any source that is the real source in the engine. This seems more like `hasSourcemap`.

jlongster commented on public/js/components/SourceFooter.js:
```diff
@@ -43,6 +44,10 @@ const SourceFooter = React.createClass({
     const { selectedSource, sourceText, togglePrettyPrint } = this.props;
     const sourceLoaded = selectedSource && !sourceText.get("loading");
 
+    if (isGenerated(selectedSource.toJS())) {
```
Don't we want to hide it for sources that have sourcemaps *and* original sources?

jasonLaster replied:
That's a good point. I change this to a selector, `hasOriginalSources`.

jasonLaster commented:
@jlongster I updated the PR and added a couple other things while in there:

+ changes the original source id to slash delineated
+ changes definition of isGenerated to not original
+ changes getGeneratedSourceId to use the source ID as opposed to
searching the map. This makes it possible to move isGenerated and
isOriginal out of the worker.

jlongster commented:
Looks great. Any reason you didn't go ahead and move them out of the worker? It's not a big deal, either way this looks good, r+

jasonLaster commented:
@jlongster no big reason. the work already seemed like it had gotten large and it took awhile to test for correctness.
